optimum/intel/openvino/configuration.py

```diff
@@ -100,31 +52,122 @@
+                f"dataset.Dataset, but found {type(self.dataset)}"
+            )
+        if not (self.ignored_scope is None or isinstance(self.ignored_scope, nncf.IgnoredScope)):
+            raise ValueError(
+                "Ignored scope must be a instance of either dict, or nncf.IgnoredScope but found "
+                f"{type(self.dataset)}"
+            )
+
+    def to_dict_without_properties(self, property_names: Union[List[str], Tuple[str]]) -> Dict[str, Any]:
+        """
+        Calls to_dict() with given properties overwritten with None. Useful for hiding non-serializable properties.
+        """
+        if len(property_names) == 0:
+            return super().to_dict()
+        with replace_properties_values(self, property_names, [None] * len(property_names)):
+            result = super().to_dict()
+        return result
+
+    def to_dict(self) -> Dict[str, Any]:
+        properties_to_omit = [] if is_serializable(self.dataset) else ["dataset"]
+        if isinstance(self.ignored_scope, nncf.IgnoredScope):
+            with replace_properties_values(self, ["ignored_scope"], [self.ignored_scope.__dict__]):
+                return self.to_dict_without_properties(properties_to_omit)
+        return self.to_dict_without_properties(properties_to_omit)
+
+
 class OVConfig(BaseConfig):
     CONFIG_NAME = "openvino_config.json"
     FULL_CONFIGURATION_FILE = "openvino_config.json"
 
     def __init__(
         self,
-        compression: Union[List[Dict], Dict, None] = None,
         input_info: Optional[List] = None,
         save_onnx_model: bool = False,
-        quantization_config: Optional[Union[QuantizationConfigMixin, Dict]] = None,
+        quantization_config: Optional[Union[dict, OVQuantizationConfigBase]] = None,
         dtype: Optional[str] = None,
         **kwargs,
     ):
         super().__init__()
-        self.compression = compression
         self.input_info = input_info
         self.save_onnx_model = save_onnx_model
-        self._enable_standard_onnx_export_option()
         self.optimum_version = kwargs.pop("optimum_version", None)
-        self.quantization_config = quantization_config or {}
+        self.quantization_config = quantization_config
+        self.compression = None  # A backward-compatability field for training-time compression parameters
 
-        if isinstance(quantization_config, QuantizationConfigMixin):
-            bits = self.quantization_config.bits
-        else:
-            bits = self.quantization_config.get("bits", None)
+        if isinstance(self.quantization_config, dict):
+            # Config is loaded as dict during deserialization
+            logger.info(
+                "`quantization_config` was provided as a dict, in this form it can't be used for quantization. "
+                "Please provide config as an instance of OVWeightQuantizationConfig or OVQuantizationConfig"
+            )
```

Comment:
why is it disabled for dict ?

Reply:
Mainly, because of the `dataset` property of the quantization config. It can't be serialized in general scenario. But if we move it to a separate parameter of `.quantize()` method as you have suggested, then I believe it will be possible.